### PR TITLE
Use screenScaleFactor to consistently adapt sizes to display pixel ratio

### PR DESCRIPTION
**Pull request by fieldOfView**
This PR attempts to solve the Hi DPI issues on both OSX and Windows. OSX apparently no longer needs the Screen.devicePixelRatio workaround when specifying pixel sizes, but Windows certainly does.

This PR is for testing an approach where instead of using Screen.devicePixelRatio, we use a Uranium-supplied screenScaleFactor. This context variable was introduced a couple of releases ago, but never used.

This PR requires https://github.com/Ultimaker/Uranium/pull/285

For additional discussion see https://github.com/Ultimaker/Cura/commit/7bc715a9bd5e61efcfef4b54f7d004b3f6248f5a#commitcomment-24539576

### CODE REVIEW

**Comment by Appesteijn** [comment]
Devs => CURA-4372

File: plugins/3MFReader/WorkspaceDialog.qml [hunk 1]
```diff
@@ -12,11 +12,13 @@ UM.Dialog
 {
     title: catalog.i18nc("@title:window", "Open Project")
 
-    width: 500
-    height: 400
+    minimumWidth: 500 * screenScaleFactor
+    minumumHeight: 400 * screenScaleFactor
```
Comment (Appesteijn) [inline plugins/3MFReader/WorkspaceDialog.qml]: Typo: minimumHeight

Comment (fieldOfView) [reply plugins/3MFReader/WorkspaceDialog.qml]: Good catch; 5659ea0

**Comment by fieldOfView** [comment]
I have included a fix for the text in the splash-screen, since that is semi-related.

**Comment by Appesteijn** [comment]
It all looks good on MacOS with a High DPI screen

**Comment by Appesteijn** [comment]
Looks good on Windows normal DPI screen, 100%. Only issue is the extruder tool top position.

**Comment by Appesteijn** [comment]
Weird things in Windows10&200%
capture
capture2
capture3
capture4
capture5

**Comment by fieldOfView** [comment]
The first two seem to be upstream issues (Qt), and are not caused by this PR. The others I will hunt down and fix.

File: resources/qml/Toolbar.qml [hunk 2]
```diff
@@ -67,7 +67,7 @@ Item
             }
         }
 
-        Item { height: UM.Theme.getSize("default_margin").height; width: 1; visible: extruders.count > 0 }
+        Item { height: UM.Theme.getSize("default_margin").height; width: UM.Theme.getSize("default_lining"); visible: extruders.count > 0 }
```
Comment (ChrisTerBeke) [inline resources/qml/Toolbar.qml]: This line causes the extruder tools to not have the correct top position. In order to fix it it should be changed to `UM.Theme.getSize("default_lining").width`. I've tested this locally but in order to simplify the merging process I'd be great if you can change that in your PR!

Comment (fieldOfView) [reply resources/qml/Toolbar.qml]: Fixed, thanks

**Comment by fieldOfView** [comment]
Has the PR been tested on Linux too?

**Comment by Appesteijn** [comment]
Not yet, working on it

**Comment by nallath** [comment]
Yup!

**Pull request merged**